pennylane/math/single_dispatch.py

```diff
@@ -200,6 +199,16 @@ def _round_tf(tensor, decimals=0):
 ar.register_function("tensorflow", "round", _round_tf)
 
 
+def _ndim_tf(tensor):
+    try:
+        return _i("tf").experimental.numpy.ndim(tensor)

```

Comment:
wait, TF doesn't have an `ndim` method?

Reply:
Ohhh I see the issue. `tf.Tensor` objects do, but `tf.Variable` objects don't :(

Reply:
Ah no, `tf.Variable` is also fine, because `tf.experimental.numpy.ndim` supports those. The problem is with `tf.function` when the input signature is provided with `TensorSpec` and a `shape=None` argument is chosen, because in this case `qml.math.ndim(input)` raises and error as `input=<unknown>`.